src/core/positioning/internalgnssreceiver.cpp

```diff
@@ -112,20 +113,36 @@ void InternalGnssReceiver::handleConnectDevice()
   {
     QLocationPermission locationPermission;
     locationPermission.setAccuracy( QLocationPermission::Precise );
-    qApp->requestPermission( locationPermission, [=]( const QPermission &permission ) {
-      if ( permission.status() == Qt::PermissionStatus::Granted )
-      {
-        mPermissionChecked = true;
-        handleConnectDevice();
-      }
-      else
-      {
-        setValid( false );
-        mLastError = tr( "Location permission denied" );
-        emit lastErrorChanged( mLastError );
-      }
-    } );
-    return;
+    Qt::PermissionStatus permissionStatus = qApp->checkPermission( locationPermission );
+    if ( permissionStatus == Qt::PermissionStatus::Undetermined )
+    {
+      // Pointer needed here to insure that the object hasn't been deleted before processing the signal
+      QPointer<InternalGnssReceiver> receiver( this );
+      qApp->requestPermission( locationPermission, [=]( const QPermission &permission ) {
+        if ( receiver )
```

Comment:
@m-kuhn , I studied the issue here and think that's an acceptable way forward. Thoughts?